packages/apollo-link-retry/src/retryLink.ts

```diff
@@ -17,6 +17,13 @@ import {
   buildRetryFunction,
 } from './retryFunction';
 
+function asPromise(result: boolean | Promise<boolean>): Promise<boolean> {
+  if (result instanceof Promise) {
```

Comment:
There are still enough folks out there using alternative `Promise` implementations that this may end up being dangerous.  I'd recommend duck typing here…

```ts
if ('then' in result && typeof result.then === 'function') {
```

…or letting TypeScript juggle it for you by converting `onError` to an `async` function, and awaiting the result of `retryIf` (you can `await` static values or promises)